src/runtime_src/core/pcie/driver/linux/xocl/xocl_drv.h

```diff
@@ -626,6 +636,12 @@ struct xocl_dev_core {
 	 * Having SN info available also implies there is a working SC
 	 */
 	char			serial_num[SERIAL_NUM_LEN];
+
+	/* XOCL Should cache all the XCLBINs which registered for this device */
+	struct xocl_axlf_obj_cache *axlf_obj[MAX_SLOT_SUPPORT];
```

Comment:
why do we use a fixed data structure (array) v.s. dynamic one (e.g., link list)? do we have performance concern here? number of registered xclbin could be > slot number, right?
xclbin with bitstream can be fairly big and will use a lot of kernel memory. register xclbin api can be called by non-privileged user. so a malicious non-root user could potentially consumes all system memory and hang the system? do we want to limit the total memory used by them instead of total number?